### PR TITLE
Update dependency management

### LINKED ISSUE #104: Update dependency management
Add dependabot, Travis, and update dependencies.

### PR DESCRIPTION
**Pull request by HaSistrunk**
Fixes #104 
Similar to [Bits & Bytes dependency updates](https://github.com/RockefellerArchiveCenter/bits_and_bytes/pull/121):

- Adds dependabot and Travis configs with specified ruby version (Travis config ignores this repo's `base` branch, which is not a gh-pages branch).
- Adds html-proofer
- Removes gemspec and uses [github-pages gem](https://github.com/github/pages-gem) in Gemfile. [Type Theme removed its own use of gemspec](https://github.com/rohanchandra/type-theme/blob/master/Gemfile) in 2019.
- The Jekyll Type Theme used here is not versioned and does not have releases, but a review of the repo commit history does not indicate changes to the theme that are relevant to our implementation.
- Updates `config.yml` to remove unused plugin and `exclude` array (based on recommendations from the [Jekyll documentation](https://jekyllrb.com/docs/troubleshooting/#configuration-problems)).

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -66,18 +66,14 @@ theme_settings:
   # Post navigation
   post_navigation: false
 
-# PAGINATION
-paginate: 5
-paginate_path: "blog/page:num"
+# # PAGINATION
+# paginate: 5
+# paginate_path: "blog/page:num"
```
Comment (helrond) [inline _config.yml]: Does it make sense to just delete these lines if we're not using this?

File: _config.yml [hunk 2]
```diff
@@ -66,18 +66,14 @@ theme_settings:
...
 highlighter: rouge
 sass:
   sass_dir: _sass
   style: :compressed
-plugins: [jekyll-paginate]
-theme: jekyll-theme-type
-exclude:
-  - Gemfile
-  - Gemfile.lock
+# plugins: [jekyll-paginate]
```
Comment (helrond) [inline _config.yml]: Same here - should we just delete?

File: _config.yml [hunk 3]
```diff
@@ -66,18 +66,14 @@ theme_settings:
...
+# # PAGINATION
+# paginate: 5
+# paginate_path: "blog/page:num"
 
 # BUILD SETTINGS
 markdown: kramdown
 highlighter: rouge
 sass:
   sass_dir: _sass
   style: :compressed
```
Comment (helrond) [inline _config.yml]:
```suggestion
  style: compressed
```

**Review by helrond — CHANGES_REQUESTED** [review]
A couple of very minor things.

**Review by helrond — APPROVED** [review]

**Review by p-galligan — APPROVED** [review]

**Review by bonniegee — APPROVED** [review]

**Pull request merged**